task/github-sarif-upload/0.1/github-sarif-upload.yaml

```diff
@@ -0,0 +1,241 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: github-sarif-upload
+  annotations:
+    tekton.dev/pipelines.minVersion: 0.12.1
+    tekton.dev/tags: image-build, konflux
+  labels:
+    app.kubernetes.io/version: "0.1"
+    build.appstudio.redhat.com/build_type: docker
+spec:
+  description: Downloads SARIF artifacts from container images and uploads them to GitHub using CodeQL CLI, making them visible in the Security tab.
+  params:
+    - name: GITHUB_REPOSITORY
+      description: GitHub repository in format 'owner/repo'
+      type: string
+    - name: GITHUB_REF
+      description: Git reference (branch or tag) for the scan results
+      type: string
+    - name: GITHUB_SHA
+      description: Git commit SHA for the scan results
+      type: string
+    - name: image-url
+      description: Image URL containing SARIF artifacts
+      type: string
+    - name: image-digest
+      description: Digest of the image containing SARIF artifacts
+      type: string
+  results:
+    - name: TEST_OUTPUT
+      description: Tekton task test output.
+  volumes:
+    - name: workdir
+      emptyDir: {}
+  stepTemplate:
+    volumeMounts:
+      - mountPath: /var/workdir
+        name: workdir
+    computeResources:
+      limits:
+        cpu: "0.5"
+        memory: 1Gi
+      requests:
+        cpu: "0.5"
+        memory: 1Gi
+    env:
+      - name: GITHUB_TOKEN
+        valueFrom:
+          secretKeyRef:
+            name: github
+            key: token
+      - name: GITHUB_REPOSITORY
+        value: $(params.GITHUB_REPOSITORY)
+      - name: GITHUB_REF
+        value: $(params.GITHUB_REF)
+      - name: GITHUB_SHA
+        value: $(params.GITHUB_SHA)
+      - name: IMAGE_URL
+        value: $(params.image-url)
+      - name: IMAGE_DIGEST
+        value: $(params.image-digest)
+  steps:
+    - name: validate-inputs
+      image: quay.io/konflux-ci/konflux-test:v1.4.40@sha256:99eb8bcc7bcb35bdd5edea7b0ac333bbdb67586dea6b4dab92baf2b8fb32bf2c
+      workingDir: /var/workdir
+      script: |
+        #!/usr/bin/env bash
+
+        set -euo pipefail
+        # shellcheck source=/dev/null
+        . /utils.sh
+        trap 'handle_error $(results.TEST_OUTPUT.path)' EXIT
+
+        echo "[$(date --utc -Ins)] Validating inputs"
+
+        # Validate required parameters
+        if [ -z "${GITHUB_TOKEN}" ]; then
+          echo "ERROR: GITHUB_TOKEN is required" >&2
+          exit 1
+        fi
+
+        if [ -z "${GITHUB_REPOSITORY}" ]; then
+          echo "ERROR: GITHUB_REPOSITORY is required" >&2
+          exit 1
+        fi
+
+        if [ -z "${GITHUB_REF}" ]; then
+          echo "ERROR: GITHUB_REF is required" >&2
+          exit 1
+        fi
+
+        if [ -z "${GITHUB_SHA}" ]; then
+          echo "ERROR: GITHUB_SHA is required" >&2
+          exit 1
+        fi
```

Comment:
Is the error checking done by Tekton not sufficient to necessitate this validation? (along with the IMAGE_URL and IMAGE_DIGEST)

Reply:
you're right. previously the task had a few optional params which needed a combination of validations. It's no longer needed. I'll get rid of it. thanks!

Reply:
Done